_episodes/03-monitor-the-model.md

```diff
@@ -0,0 +1,591 @@
+> > Trainable params: 21,501
+> > Non-trainable params: 0
+> > _________________________________________________________________
+> > ~~~
+> > {:.output}
+> >
+> > The shape of the input layer has to correspond to the number of features in our data: 163
+> > 
+> > The output layer here is a dense layer with only 1 node. And we here have chosen to use *no activation function*.
+> > While we might use *softmax* for a classification task, here we do not want to restrict the possible outcomes for a start.
+> > 
+> > In addition, we have here chosen to write the network creation as a function so that we can use it later again to initiate new models.
+> {:.solution}
+{:.challenge}
+
+When compiling the model we can define a few very important aspects.
+
+### Loss function:
+The loss is what the neural network will be optimized on during training, so chosing a suitable loss function is crucial for training neural networks.
+In the given case we want to stimulate that the prodicted values are as close as possible to the true values. This is commonly done by using the *mean squared error* (mse) or the *mean absolute error* (mae), both of which should work OK in this case. Often, mse is prefered over mae because it "punishes" large prediction errors more severely.
+In keras this is implemented in the `keras.losses.MeanSquaredError` class (see keras documentation: https://keras.io/api/losses/).
+
+### Optimizer:
+Somewhat coupled to the loss function is the *optimizer* that we want to use. 
+The *optimizer* here refers to the algorithm with which the model learns to optimize on the set loss function. A basic example for such an optimizer would be *stochastic gradient descent*. For now, we can largely skip this step and simply pick one of the most common optimizers that works well for most tasks: the *Adam optimizer*. 
+
+### Metrics:
+In our first example (episode 2) we plotted the progression of the loss during training. 
+That is indeed a good first indicator if things are working alright, i.e. if the loss is indeed decreasing as it should. 
+However, when models become more complicated then also the loss functions often become less intuitive (side remark: e.g. when adding L1 or L2 regularization). 
+That is why it is good practice to monitor the training process with additional, more intuitive metrics. 
+They are not used to optimize the model, but are simply recorded during training. 
+With Keras they can simply be added via `metrics=[...]` and can contain one or multiple metrics of interest. 
+Here we could for instance chose to use `'mae'` the mean absolute error, or the the *root mean squared error* (RMSE) which unlike the *mse* has the same units as the predicted values.
+
+~~~
+model.compile(optimizer='adam',
+              loss='mse',
+              metrics=[keras.metrics.RootMeanSquaredError()])
+~~~
+{: .language-python}
+
+## Train a dense neural network
+Now that we created and compiled our dense neural network, we can start training it.
```

Comment:
Maybe this would be a good point to explain what `batch_size` means. The concept of training batch is not explained in episode 1 and 2. It seems like an important concept to explain, also in order to understand batch normalization. This should probably also be added to the learning objectives.

Reply:
I added a sentence or two on it here. But I feel that might be a bit much for some... let's re-evaluate that and see where to introduce it in a proper way. I feel like gradient descent, learning rate, and batch-size all belong together and would benefit from being introduced together with a 3D loss-landscape plot.

Reply:
I agree, I created an issue #106